src/sinks/datadog/metrics/config.rs

```diff
@@ -92,37 +92,50 @@ impl DatadogMetricsEndpointConfiguration {
 
 /// Configuration for the `datadog_metrics` sink.
 #[configurable_component(sink("datadog_metrics"))]
-#[derive(Clone, Debug, Default)]
+#[derive(Clone, Debug)]
 #[serde(deny_unknown_fields)]
 pub struct DatadogMetricsConfig {
     /// Sets the default namespace for any metrics sent.
     ///
     /// This namespace is only used if a metric has no existing namespace. When a namespace is
     /// present, it is used as a prefix to the metric name, and separated with a period (`.`).
+    #[configurable(metadata(docs::examples = "myservice"))]
+    #[serde(default)]
     pub default_namespace: Option<String>,
 
     /// The endpoint to send metrics to.
+    ///
+    /// The endpoint must contain an HTTP scheme, and may specify a
+    /// hostname or IP address and port.
+    ///
+    /// If set, overrides the `site` option.
+    #[configurable(metadata(docs::examples = "http://127.0.0.1:8080"))]
+    #[configurable(metadata(docs::examples = "http://example.com:12345"))]
+    #[serde(default)]
     pub(crate) endpoint: Option<String>,
 
     /// The Datadog region to send metrics to.
-    ///
-    /// This option is deprecated, and the `site` field should be used instead.
-    #[configurable(deprecated)]
+    #[configurable(deprecated = "This option has been deprecated, use the `site` option instead.")]
+    #[serde(default)]
     pub region: Option<Region>,
 
     /// The Datadog [site][dd_site] to send metrics to.
     ///
     /// [dd_site]: https://docs.datadoghq.com/getting_started/site
     #[serde(default = "default_site")]
+    #[configurable(metadata(docs::examples = "us3.datadoghq.com"))]
+    #[configurable(metadata(docs::examples = "datadoghq.eu"))]
     pub site: String,
```

Comment:
**thought**: As the possible sites are known, they could be an enum to return improved user feedback on misconfigurations.

Reply:
The main issue I can conceive with that, is that if they add a new site, how would we know to update that enum 🤔 

To get us closer to that, I could add the other current sites as examples ?

Reply:
Sure, maybe not necessary - lazy answer could be linking out to the DD docs that list all the sites (if that exists).

Definitely a non-blocking comment - just came up as I'd been thinking a lot about improving the user experience

Reply:
I like the idea. I think I'll tackle that in the mini refactor, so don't have to edit each branch.

Reply:
Sounds good to me!